semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbTaskListener.java

```diff
@@ -104,13 +122,84 @@ private void onFinishedAnalyze(TaskEvent e) {
   }
 
   private void writeSemanticdb(TaskEvent event, Path output, Semanticdb.TextDocument textDocument) {
-    try {
-      byte[] bytes =
-          Semanticdb.TextDocuments.newBuilder().addDocuments(textDocument).build().toByteArray();
-      Files.createDirectories(output.getParent());
-      Files.write(output, bytes);
-    } catch (IOException e) {
-      this.reportException(e, event);
+    if (Files.exists(output)) {
```

Comment:
Nit: it would be nice to extract this condition into a separate function so that `writeSemanticdb` can focus on just writing to disk, not doing merging.